src/ring_jwt_middleware/result.clj

```diff
@@ -0,0 +1,93 @@
+(ns ring-jwt-middleware.result
+  "This ns provide a set of helpers to handle an abstraction similar to Either in Haskell
+
+  The main goal is to provide a mechanism similar to the exceptions but pure without Java Exceptions.
+
+  A function that return a `Result` means the function returned either a successful result or an error with a
+  common error structure (`JwtError`).
+
+  The `let-either` macro provides a monadic syntax.
+  Mainly:
+
+  ```
+  (let-either [result-value-1 (fn-returning-a-result-1 ,,,)
+               result-value-2 (fn-returning-a-result-2 ,,,)
+               ,,,]
+    ,,,)
+  ```
+
+  if `fn-returning-a-result-1` return an error then we will not execute the rest of the let-either.
+  And return the full `result`.
+  "
+  (:require [schema.core :as s]
+            [schema-tools.core :as st]))
+
+(s/defschema JwtError
+  (st/open-schema
+    {:error s/Keyword
+     :error_description s/Str}))
+
+(s/defn result-of
+  "Build a schema representing a result expecting succesful result with schema `s`"
+  [s]
+  {(s/optional-key :result) s
+   (s/optional-key :jwt-error) JwtError})
+
+(s/defschema Result
+  ;; A result is similar to the Either in Haskell
+  ;; It represent either a value or an error
```

Comment:
It looks like s/defschema takes docstring. Maybe you meant this comment to be a docstring?